src/renderer/react/app/nodes/editNodeModal/EditNodeModal.jsx

```diff
@@ -1,47 +1,51 @@
 import { noop, mapValues } from 'lodash';
 import React from 'react';
 import PropTypes from 'prop-types';
-import { compose } from 'recompose';
+import { compose, withPropsOnChange } from 'recompose';
 import { createForm, createFormField } from 'rc-form';
-import { createSelector } from 'reselect';
+import { FormattedMessage } from 'react-intl';
+import { connect } from 'react-redux';
 
 import { listDescendants } from 'services/node';
 import { nodeShape } from 'react/shapes/node';
 import { formShape } from 'react/shapes/form';
+import Button from 'react/components/general/button/Button';
+import globalMessages from 'config/global.messages';
+import { getNodesMap } from 'redux/stores/nodes/selector';
 
-/**
- * Maps props to a rc-form friendly default value.
- *
- * @param {object} props - Props.
- * @returns {object} - Values.
- */
-const defaultValueMapper
-  = props => mapValues(props.node || {}, value => createFormField({ value }));
+const enhancer = compose(
+  connect(state => ({
+    nodesMap: getNodesMap(state),
+  }), null),
+
+  /**
+   * Make a list of available parents and format it for select/options.
+   */
+  withPropsOnChange(
```

Comment:
Is there any reason why you didn't just extract your selector in the store?

Reply:
And selecting it in your connect above next to the `nodesMap`.

Reply:
Because this form was a pain in the ass to refactor and I left in the middle of it lol